src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -68,7 +68,11 @@ public abstract class NodeModel : ModelBase, IRenderPackageSource<NodeModel>, ID
 
         private readonly Dictionary<int, Tuple<int, NodeModel>> inputNodes;
         private readonly Dictionary<int, HashSet<Tuple<int, NodeModel>>> outputNodes;
-
+        /// <summary>
+        /// Can be used to suppress some property notifications related to node warnings/state. This is useful
+        /// to defer UI updates for better performance. //TODO should this be generic for all nodemodel props?
+        /// </summary>
+        internal bool SuppressStatePropertyNotifications { get; set; } = false;
```

Comment:
From reading your description, it almost seems that this flag will never be true? Do we want to ask Matterlab to update the design of that feature?

Reply:
this flag will be `true` only when when we're en masse setting the `NodeModel.ToolTipText` property or `state` -  at other times it will be set false so that if a client application (for example an extension) sets a single node's state the UI should update automatically.

We're only taking control and making these updates explicit when runs are occurring and it's likely theres going to be many updates happening.

If we want to improve performance for all clients with arbitrarily large change sets we'll need to rethink dynamo's UI architecture.

Reply:
tihs flag could be moved to the notificationObject class